### PR TITLE
[PM-8210] Reintroduce Foreground & Background Derived State

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Fix a race condition in the retrieval of decrypted ciphers while they are being updated. Inlined derived state didn't respect the `forceValue` method much at all. The thinking was it doesn't have it's own cache so there isn't a security reason to force a value. But `CipherService` is using it to block emissions during an update so I broke this. Adding it back because the original bug that caused us to remove it seems to have gotten fixed in another way. But I will have QA do full automation to make sure. 

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/browser/src/platform/state/background-derived-state.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  }
+
+  private async sendResponse(
+    originalMessage: DerivedStateMessage,
+    response: Omit<DerivedStateMessage, "originator" | "id">,
+    port: chrome.runtime.Port,
+  ) {
+    // FIXME: Verify that this floating promise is intentional. If it is, add an explanatory comment and ensure there is proper error handling.
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.sendMessage(
```
Comment (MGibson1) [inline apps/browser/src/platform/state/background-derived-state.ts:89]:
```suggestion
    await this.sendMessage(
```

**Review by MGibson1 — CHANGES_REQUESTED** [review]
small but hopefully meaningful changes

File: apps/browser/src/platform/state/foreground-derived-state.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+export class ForegroundDerivedState<TTo> implements DerivedState<TTo> {
+  private port: chrome.runtime.Port;
+  private backgroundResponses$: Observable<DerivedStateMessage>;
+  state$: Observable<TTo>;
+
+  constructor(
+    private deriveDefinition: DeriveDefinition<unknown, TTo, DerivedStateDependencies>,
+    private portName: string,
+    private pipeCustomizer: OperatorFunction<unknown, TTo>,
```
[the remark this answers is not in the document]
Comment (justindbaur) [reply apps/browser/src/platform/state/foreground-derived-state.ts]: Good call, this is definitely not supposed to make any changes to the data.

File: apps/browser/src/platform/state/background-derived-state.ts (hunk 1 above)
Comment (justindbaur) [reply apps/browser/src/platform/state/background-derived-state.ts]: Was going to do this but `sendMessage` didn't actually need to be async and therefore `sendResponse` didn't need to be either so I removed that instead.

**Review by Jingo88 — APPROVED** [review]

**Pull request closed without merging**